x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,728 @@
+# RFC: Prebuilt Rules Customization
+
+## Necessary rule schema changes
+
+In order to support the customization of Elastic Prebuilt Rules, we need to modify our rule schema. This involves introducing two new fields: `prebuilt` and `customized`. Simultaneously, we need to deprecate the existing `immutable` field, which determines if a rule is an Elastic Prebuilt Rule and whether its fields can be modified/customized.
+
+To ensure backward compatibility and avoid breaking changes, we will silently deprecate the `immutable` field but keep it unchanged (i.e. non-optional).
+
+Since the migration of rules will be performed incrementally (see below), we have to assume that, at any given point in time, a user may have a mix of migrated and non-migrated rule saved objects. Consequently, we must continue supporting both versions while also carrying out the logic for migrating rules to the new schema.
+
+In summary, the changes to be done to our rule schema are:
+
+- Mark the `immutable` field as deprecated but keep unchanged for backwards compatibility.
+    - Should remain a `boolean` type and non-optional (`required`)
+- Add `prebuilt` field to the rule schema
+    - `prebuilt` should have same type as `immutable`: `boolean`
+    - Should be non-optional (`required`)
+    - During migration, set the `prebuilt` value to match the current value of `immutable`. This means: prebuilt rules should be migrated with a `prebuilt` value set to `true`, while custom rules should be set with a `prebuilt` value of `false`.
```

Comment:
Would it be correct to say that on our API surface we will maintain the following invariant:

```ts
const alwaysTrue = rule.immutable === rule.prebuilt;
```

Reply:
Yes, clarified this in this section.